src/UI/App.jsx

```diff
@@ -594,6 +595,22 @@ export default class App extends React.Component {
 		} else if(path[layer0] === "authHeader") {
 			objectToSaveIntoState = {deviceAuthorization: value};
 		} else if(path[layer0] === "weekdays") {
+			// Find existing weekdays that have the same day name as the new value
+			// If we find any we delete them so the new setting can override them.
+			value.forEach((newWeekday) => {
+				const existingWeekday = this.state.workingHours?.find((stateWeekday) => {
+					if(stateWeekday[0].toLowerCase !== undefined) {
+						// Dealing with ["Day", start, end]
+						return stateWeekday[0].toLowerCase() === newWeekday[0].toLowerCase();
+					}
+
+					// Dealing with [startTimestamp, endTimestamp]
+					return stateWeekday[0] === newWeekday[0];
+				});
+				if(existingWeekday)
+					this.state.workingHours.splice(this.state.workingHours.indexOf(existingWeekday), 1);
```

Comment:
Hmm ik heb geleerd dat je setState moet gebruiken om de state aan te passen en niet zoals hier.
Zie:
https://legacy.reactjs.org/docs/state-and-lifecycle.html#using-state-correctly

Maar ik zie wel dat setState altijd een re-render triggerd en als dat niet wenselijk is snap ik het

Reply:
setState wordt verder onder gedaan: https://github.com/parley-messaging/web-library/pull/59/files/af2e14b18bb6395c006e838422493c878748b33e#diff-0abef088b3f6260f845818ba7eef7e55a43e70e212c423ebaa54066f4c94aba7R640
ik wil dat hier niet doen omdat dat idd weer andere dingen triggerd.

Reply:
Hmm moet je dan niet working hours in een let zetten en die mee geven aan de setState? dus;
```js
let workingHours = this.state.workingHourse;
workingHours.splice(this.state.workingHours.indexOf(existingWeekday), 1);
obectToSaveIntoState.workingHours = workingHours;
this.setState(prevState => deepMerge(prevState, objectToSaveIntoState));
```
zo iets?

Reply:
`deepMerge` zorgt er voor dat hij dan niet verwijderd wordt helaas..